pkgs/development/libraries/science/math/dealii/default.nix

```diff
@@ -0,0 +1,183 @@
+  };
+
+  nativeBuildInputs = [ cmake doxygen perl graphviz ];
+  buildInputs = [ boostJoined hdf5Joined zlibJoined blas lapack ]
+    ++ lib.optionals withThreading [ tbb ]
+    ++ lib.optionals withMPI [ mpi metis p4est p4est-sc ]
+    ++ lib.optionals petscEnabled [ petsc ]
+    ++ lib.optionals (!minimal) [
+      arpack assimp gmsh gsl muparser suitesparse symengine scalapack
+    ];
+  checkInputs = lib.optionals (withMPI && mpi.pname == "openmpi") [ openssh ];
+
+  outputs = [ "out" "doc" ];
+
+  cmakeFlags = [
+      "-DCMAKE_BUILD_TYPE=${buildType}"
+      "-DDEAL_II_ALLOW_BUNDLED=OFF"
+      "-DDEAL_II_ALLOW_AUTODETECTION=OFF"
+      "-DDEAL_II_WITH_BOOST=ON" "-DBOOST_DIR=${boostJoined}"
+      "-DDEAL_II_WITH_ZLIB=ON" "-DZLIB_DIR=${zlibJoined}"
+      "-DDEAL_II_WITH_HDF5=ON" "-DHDF5_DIR=${hdf5Joined}"
+      "-DBLAS_DIR=${blas}"
+      "-DDEAL_II_WITH_LAPACK=ON" "-DLAPACK_DIR=${lapack}"
+      "-DDEAL_II_COMPONENT_DOCUMENTATION=ON"
+      "-DDEAL_II_COMPONENT_EXAMPLES=OFF"
+      "-DDEAL_II_DOXYGEN_USE_MATHJAX=ON"
+      # deal.II needs mathjax 2, which isn't available in Nixpkgs
+      "-DDEAL_II_DOXYGEN_USE_ONLINE_MATHJAX=ON"
+    ]
+    ++ lib.optionals withThreading  [
+      "-DDEAL_II_WITH_TBB=ON" "-DTBB_DIR=${tbb}"
+    ]
+    ++ lib.optionals withMPI [
+      "-DDEAL_II_WITH_MPI=ON"
+      "-DDEAL_II_WITH_METIS=ON" "-DMETIS_DIR=${metis}"
+      "-DDEAL_II_WITH_P4EST=ON" "-DP4EST_DIR=${p4est}" "-DSC_DIR=${p4est-sc}"
+    ]
+    ++ lib.optionals with64bitIndices [ "-DDEAL_II_WITH_64BIT_INDICES=ON" ]
+    ++ lib.optionals petscEnabled [
+      "-DDEAL_II_WITH_PETSC=ON" "-DPETSC_DIR=${petsc}"
+    ]
+    ++ lib.optionals (!minimal) [
+      "-DDEAL_II_WITH_MUPARSER=ON" "-DMUPARSER_DIR=${muparser}"
+      "-DDEAL_II_WITH_ARPACK=ON" "-DARPACK_DIR=${arpack}"
+      "-DDEAL_II_WITH_ASSIMP=ON" "-DASSIMP_DIR=${assimp}"
+      "-DDEAL_II_WITH_GMSH=ON" "-DGMSH_DIR=${gmsh}"
+      "-DDEAL_II_WITH_GSL=ON" "-DGSL_DIR=${gsl}"
+      "-DDEAL_II_WITH_SCALAPACK=ON" "-DSCALAPACK_DIR=${scalapack}"
+      "-DDEAL_II_WITH_SYMENGINE=ON" "-DSYMENGINE_DIR=${symengine}"
+      "-DDEAL_II_WITH_UMFPACK=ON" "-DUMFPACK_DIR=${suitesparse}"
+    ];
+
+  postPatch = ''
+    for s in /build/source/doc/doxygen/scripts/*; do
+      patchShebangs "$s";
+    done
+  '';
+
+  # See https://discourse.nixos.org/t/cmakeflags-and-spaces-in-option-values/20170
+  preConfigure = lib.optionalString (commonCxxFlags != "") ''
+      cmakeFlagsArray+=(
+        "-DDEAL_II_CXX_FLAGS=${commonCxxFlags}"
+      )
+    ''
+    + lib.optionalString (debugCxxFlags != "") ''
+      cmakeFlagsArray+=(
+        "-DDEAL_II_CXX_FLAGS_DEBUG=${debugCxxFlags}"
+      )
+    ''
+    + lib.optionalString (releaseCxxFlags != "") ''
+      cmakeFlagsArray+=(
+        "-DDEAL_II_CXX_FLAGS_RELEASE=${releaseCxxFlags}"
+      )
+    '';
+
+  postBuild = ''
+    make documentation
+    cp -r ${offlineDocs}/doc/doxygen/deal.II/images /build/source/build/doc/doxygen/deal.II
+  '';
```

Comment:
Using the absolute path in the sandbox is not allowed.

Reply:
Thanks, will fix.